trino/dbapi.py

```diff
@@ -228,10 +232,28 @@ def cursor(self, legacy_primitive_types: bool = None):
         return Cursor(
             self,
             request,
-            # if legacy_primitive_types is not explicitly set in Cursor, take from Connection
+            # if legacy params are not explicitly set in Cursor, take them from Connection
             legacy_primitive_types if legacy_primitive_types is not None else self.legacy_primitive_types
         )
 
+    @functools.lru_cache(maxsize=512)
```

Comment:
Adding `@functools.lru_cache` on an object instance method will leak memory. as the instance can't be garbage collected.

Reply:
Thank you for catching that.